test/optimization/test_converters.py

```diff
@@ -40,6 +65,46 @@ def test_empty_problem(self):
         qubitop, shift = conv.encode(op)
         self.assertEqual(shift, 0.0)
 
+    def test_valid_variable_type(self):
+        # validate the types of the variables for OptimizationProblemToOperator
+        # Integer variable
+        with self.assertRaises(QiskitOptimizationError):
+            op = OptimizationProblem()
+            op.variables.add(names=['x'], types='I')
+            conv = OptimizationProblemToOperator()
+            op2 = conv.encode(op)
```

Comment:
Pylint will complain about the unused variable
```suggestion
            _ = conv.encode(op)
```